spec/datadog/tracing/contrib/propagation/sql_comment_spec.rb

```diff
@@ -79,6 +78,7 @@
         let(:mode) { 'service' }
 
         it do
+          allow(span_op).to receive(:get_tag).with('peer.service').and_return(nil)
```

Comment:
The test was simple enough so we could use a mock object. Since it is becoming more complex, replace `span_op` from mock object with a real object like [this](https://github.com/DataDog/dd-trace-rb/pull/3127/files#diff-8ac548c14d9fa2993b3cdc2768cf4ff4dc4be995602c7daf448b03dee7477d76R8)